hare/hare.go

```diff
@@ -174,8 +173,8 @@ func (h *Hare) HandleEligibility(ctx context.Context, emsg *types.HareEligibilit
 }
 
 func (h *Hare) getLastLayer() types.LayerID {
-	h.layerLock.RLock()
-	defer h.layerLock.RUnlock()
+	h.mu.Lock()
+	defer h.mu.Unlock()
```

Comment:
Why did you change the `RWMutex` to a `Mutex`? It made sense to lock for reading only here.

Reply:
mostly because i prefer code simplicity over premature optimization. 

- Mutex is more straightforward to use
- RWMutex.Lock is slower than Mutex.Lock
- in hare module, the only place one needs to worry about thread-safety is the gossip handler. the rest should not have lock contention

i'd use RWMutex only when the component is write-once and read-many-times. otherwise i prefer simplicity.